e2e/testsuite/codec.go

```diff
@@ -64,3 +70,29 @@ func codecAndEncodingConfig() (*codec.ProtoCodec, simappparams.EncodingConfig) {
 	cdc := codec.NewProtoCodec(cfg.InterfaceRegistry)
 	return cdc, cfg
 }
+
+// UnmarshalMsgResponses attempts to unmarshal the tx msg responses into the provided message types.
+func UnmarshalMsgResponses(txResp sdk.TxResponse, msgs ...codec.ProtoMarshaler) error {
```

Comment:
pretty happy with how this function turned out 🥇

Reply:
Yeah, its pretty nice! :D Hopefully will become useful in future too